nbodykit/base/mesh.py

```diff
@@ -34,44 +34,67 @@ def __init__(self, comm, Nmesh, BoxSize, dtype):
 
         # ensure self.comm is set, though usually already set by the child.
         self.comm = comm
-
         self.dtype = dtype
-        Nmesh = numpy.array(Nmesh)
-        if Nmesh.ndim == 0:
-            ndim = 3
-        else:
-            ndim = len(Nmesh)
-        _Nmesh = numpy.empty(ndim, dtype='i8')
+        self.attrs['Nmesh'] = Nmesh
+        self.attrs['BoxSize'] = BoxSize
+
+        # initialize the ParticleMesh
+        initialize_pm(self)
+
+        self._actions = []
+        self.basemesh = None
 
-        _Nmesh[:] = Nmesh
-        self.pm = ParticleMesh(BoxSize=BoxSize,
-                                Nmesh=_Nmesh,
-                                dtype=self.dtype, comm=self.comm)
+    def __finalize__(self, obj):
+        """
+        Return a "view" of the MeshSource, in the spirit of
+        numpy's ndarray view.
+
+        This returns a new MeshSource whose memory is owned by ``self``.
+        """
+        self.comm = obj.comm
+        self.dtype = obj.dtype
+        self.attrs.update(obj.attrs)
 
-        self.attrs['BoxSize'] = self.pm.BoxSize.copy()
-        self.attrs['Nmesh'] = self.pm.Nmesh.copy()
+        if hasattr(obj, 'pm'):
+            self.pm = obj.pm
+        else:
+            initialize_pm(self)
 
         self._actions = []
-        self.base = None
+        self.actions.extend(getattr(obj, 'actions', []))
```

Comment:
This is heading towards the direction of having 'actions' as a part of dask diagram..

Reply:
How so? You can't have actions applied after the ``r2c`` that way?

Reply:
More or less like dask.delayed?